includes/CLI/Plugin_Check_Command.php

```diff
@@ -0,0 +1,312 @@
+<?php
+/**
+ * Class WordPress\Plugin_Check\CLI\Plugin_Check_Command
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\CLI;
+
+use WordPress\Plugin_Check\Plugin_Context;
+use WordPress\Plugin_Check\Utilities\Plugin_Request_Utility;
+use WordPress\Plugin_Check\Checker\CLI_Runner;
+use Exception;
+use WP_CLI;
+
+/**
+ * Plugin check command.
+ */
+class Plugin_Check_Command {
+
+	/**
+	 * Plugin context.
+	 *
+	 * @since n.e.x.t
+	 * @var Plugin_Context
+	 */
+	protected $plugin_context;
+
+	/**
+	 * Output format type.
+	 *
+	 * @since n.e.x.t
+	 * @var string[]
+	 */
+	protected $output_formats = array(
+		'table',
+		'csv',
+		'json',
+	);
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Plugin_Context $plugin_context Plugin context.
+	 */
+	public function __construct( Plugin_Context $plugin_context ) {
+		$this->plugin_context = $plugin_context;
+	}
+
+	/**
+	 * Runs plugin check.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <plugin>
+	 * : The plugin to check. Plugin name.
+	 *
+	 * [--checks=<checks>]
+	 * : Only runs checks provided as an argument in comma-separated values, e.g. enqueued-scripts, escaping. Otherwise runs all checks.
+	 *
+	 * [--format=<format>]
+	 * : Format to display the results. Options are table, csv, and json. The default will be a table.
+	 * ---
+	 * default: table
+	 * options:
+	 *   - table
+	 *   - csv
+	 *   - json
+	 * ---
+	 *
+	 * [--fields=<fields>]
+	 * : Limit displayed results to a subset of fields provided.
+	 *
+	 * [--ignore-warnings]
+	 * : Limit displayed results to exclude warnings.
+	 *
+	 * [--ignore-errors]
+	 * : Limit displayed results to exclude errors.
+	 *
+	 *
+	 * ## EXAMPLES
+	 *
+	 *   wp plugin check akismet
+	 *   wp plugin check akismet --checks=escaping
+	 *   wp plugin check akismet --format=json
+	 *
+	 * @subcommand check
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $args       List of the positional arguments.
+	 * @param array $assoc_args List of the associative arguments.
+	 *
+	 * @throws Exception Throws exception.
+	 */
+	public function check( $args, $assoc_args ) {
+		// Get options based on the CLI arguments.
+		$options = $this->get_options( $assoc_args );
+
+		// Create the plugin and checks array from CLI arguments.
+		$plugin = isset( $args[0] ) ? $args[0] : '';
+		$checks = wp_parse_list( $options['checks'] );
+
+		// Get the CLI Runner.
+		$runner = Plugin_Request_Utility::get_runner();
+
+		if ( is_null( $runner ) ) {
+			$runner = new CLI_Runner();
+			$runner->set_plugin( $plugin );
+			$runner->set_check_slugs( $checks );
```

Comment:
These calls need to happen unconditionally, even if you're using the runner returned by `Plugin_Request_Utility::get_runner`. That is the entire reason why we implemented the validation in `Abstract_Check_Runner`.